### ISSUE TITLE
MDVA-20376 bug

### ISSUE THREAD

**Issue by webtekindo** [magento/quality-patches#49]
Magento 2.3.3

**Configuration**
Install patch MDVA-20376
Enable Persistence = Yes
Clear Persistence on Sign Out = No
Persist Shopping Cart = Yes

**Steps**

1) As a guest add product to shopping cart
2) Login with existing customer account
3) Logout

**Results**
Mini cart is empty and Shopping cart is not emtpy

**Expected results**
Mini cart is not empty and shopping cart not empty

You can also see here:

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Model/Session.php#L289

That it was fixed.

`$this->setQuoteId($quote->getId());`

should be

`$this->setQuoteId($quoteByCustomer->getId());`

**Comment by ihor-sviziev** [comment]
@webtekindo, could you provide more info on what Magento version you are using, how to reproduce the issue, what is actual and expected results?

**Comment by webtekindo** [comment]
@ihor-sviziev I already update the issue description.

**Comment by ihor-sviziev** [comment]
Looks like this issue relates to https://github.com/magento/magento2/pull/25307

**Comment by webtekindo** [comment]
Yes correct, the patch does the same thing.

Just this line is incorrect in the patch.

https://github.com/magento/magento2/pull/25307/files#diff-ddbd0579d2fe16ac5f44a5e2b03fa586aa84aa626665ec324749548f05f16bf8R284

**Comment by viktym** [comment]
@webtekindo thanks for your report
@Rykh can you take a look?

**Comment by Rykh** [comment]
A patch MDVA-20376 fixes the issue with the 'No such entity with customerId = 1' for logged-in customers. 
The fix from a patch MDVA-20376 does not affect line 289 (see https://github.com/magento/quality-patches/blob/master/patches/os/MDVA-20376_2.3.2.patch#L29). This patch based on a fix from https://github.com/magento/magento2/pull/23952. The fix $this->setQuoteId($quoteByCustomer->getId()); was added in another PR.
It will be great if you can contribute this backport to our MQP tool as a separate patch dependent on MDVA-20376. We will include it into next releases.

**Comment by hostep** [comment]
@Rykh: actually https://github.com/magento/magento2/pull/23952 which you refer to never got accepted and was replaced with https://github.com/magento/magento2/pull/25307

I would suggest you to replace that patch in this repo as well, and not make a new patch, what do you think?

**Comment by Rykh** [comment]
MDVA-20376 was tested and resolve the described issue. 
I added this change request to our backlog. I am not sure about it priority and time when we can deliver this fix. 

You can deliver this fix into MQP if this issue affects your Magento instance.

**Comment by hostep** [comment]
It might have been tested manually, but the automated tests actually discovered the issue that @webtekindo mentions here, see my comment from https://github.com/magento/magento2/pull/25307#issuecomment-546694686

**Comment by viktym** [comment]
@hostep @ihor-sviziev the patch MDVA-20376 will be replaced with improved implementation in the next release _magento/quality-patches:1.0.21_
Thanks for reporting

**Comment by viktym** [comment]
Patch MDVA-20376 was replaced with MDVA-25785
